### PR TITLE
Simple improvements (for a developer) to run routine-connector.

### PR DESCRIPTION
**Pull request by mmwaikar**
Use dotenv crate to read from .env file.
Use pretty_assertions create to highlight where a test has failed.
Add an example-dotenv file (from where the key-value pairs can be copied)
Improve tests (call unwrap only if the result is ok).

### CODE REVIEW

**Review by torbrenner — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ edition = "2021"
 axum = "0.7.5"
 chrono = { version = "0.4.37", default-features = false, features = ["serde", "now"] }
 clap = { version = "4.5.3", features = ["env", "derive"] }
+dotenv = "0.15.0"
```
Comment (Threated) [inline Cargo.toml]: This crate has not seen updates in 4 years the maintained version of it is https://crates.io/crates/dotenvy

File: src/main.rs [hunk 2]
```diff
@@ -135,32 +138,37 @@ async fn fetch_data() -> Result<String, String> {
...
-        let response = reqwest::Client::new()
+        let response_result = reqwest::Client::new()
             .post(format!("http://localhost:8080/requests"))
             .json(&serde_json::from_slice::<serde_json::Value>(json).unwrap())
             .send()
-            .await
-            .unwrap();
-        assert_eq!(response.status(), StatusCode::CREATED);
+            .await;
+        assert_eq!(response_result.is_ok(), true, "POST response (from /requests) was not OK");
```
Comment (Threated) [inline src/main.rs]: I dont really get how this improves the code. Unwraping/panicing in tests is totally fine. This assert will just discard the actual error message. If you want to add a more descriptive error message you could use `Result::expect`.

File: src/main.rs [hunk 3]
```diff
@@ -135,32 +138,37 @@ async fn fetch_data() -> Result<String, String> {
...
+
+        let response_result = reqwest::Client::new()
+            .get(url)
             .send()
-            .await
-            .unwrap();
+            .await;
+        assert!(response_result.is_ok(), "GET response (from /requests/id) was not OK");
 
+        let response = response_result.unwrap();
```
Comment (Threated) [inline src/main.rs]: Same as above. Also while you are at it you can remove the dbg macro

**Review by Threated — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (mmwaikar) [reply Cargo.toml]: Done.

File: src/main.rs (hunk 2 above)
Comment (mmwaikar) [reply src/main.rs]: Done.

File: src/main.rs (hunk 3 above)
Comment (mmwaikar) [reply src/main.rs]: Done, removed dbg macro.

**Comment by Threated** [comment]
Maybe we can avoid dotenv entirely by using this https://doc.rust-lang.org/nightly/cargo/reference/config.html#env

File: example-dotenv.md
Comment (Threated) [inline example-dotenv.md:1]: Do we need this file now that we don't use dotenvy anymore

File: src/main.rs [hunk 4]
```diff
@@ -135,33 +135,41 @@ async fn fetch_data() -> Result<String, String> {
...
     use crate::requests::DataRequest;
 
-    async fn post_data_request() -> DataRequest {
-        let json = include_bytes!("../docs/examples/data_request.json");
+    async fn post_data_request() -> DataRequest {        
+        let bytes = include_bytes!("../docs/examples/data_request.json");        
+        let json = &serde_json::from_slice::<serde_json::Value>(bytes).unwrap();
+
+        // NOTE: this test fails if the no_proxy env var is not set, hence this dbg stmt
+        dbg!(env::var("no_proxy").is_ok());
```
Comment (Threated) [inline src/main.rs]: `reqwest` uses the no proxy env var by default. Besides that this line only prints whether `no_proxy` is set which is not that useful I feel like

File: src/main.rs [hunk 5]
```diff
@@ -135,33 +135,41 @@ async fn fetch_data() -> Result<String, String> {
...
             .await
-            .unwrap();
+            .expect("POST endpoint (/requests) should give a valid response");
         assert_eq!(response.status(), StatusCode::CREATED);
-
         response.json().await.unwrap()
     }
 
     #[tokio::test]
+    // #[traced_test]
```
Comment (Threated) [inline src/main.rs]:
```suggestion
```

**Review by Threated — CHANGES_REQUESTED** [review]
Also can you commit the .cargo/config.toml

File: example-dotenv.md
Comment (mmwaikar) [reply example-dotenv.md:1]: Deleted.

File: src/main.rs (hunk 4 above)
Comment (mmwaikar) [reply src/main.rs]: Deleted.

File: src/main.rs (hunk 5 above)
Comment (mmwaikar) [reply src/main.rs]: Deleted.

**Pull request merged**